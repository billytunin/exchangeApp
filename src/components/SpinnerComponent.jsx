import React, { Component } from 'react';
import '../css/SpinnerComponent.css';

class SpinnerComponent extends Component {
  render() {
    return(
      <div className={`spinner_container ${this.props.show ? 'show' : 'hide'}`}>
        <div className="spinner"></div>
      </div>
    )
  }
}

export default SpinnerComponent;
