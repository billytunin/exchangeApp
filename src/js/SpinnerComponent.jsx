import React, { Component } from 'react';
import '../css/SpinnerComponent.css';

class SpinnerComponent extends Component {
  render() {
    
  }
}

export default SpinnerComponent;
