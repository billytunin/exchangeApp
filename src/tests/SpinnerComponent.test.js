import React from 'react';
import SpinnerComponent from '../components/SpinnerComponent.jsx';
import renderer from 'react-test-renderer';

describe('Suite de tests del componente SpinnerComponent', () => {

  it('renderiza el componente sin errores y con las props iniciales', () => {
    const component = renderer.create(<SpinnerComponent show={true}></SpinnerComponent>);
    const componentInstance = component.getInstance();

    expect(componentInstance.props.show).toEqual(true);
  });

  it('coincide con el Ãºltimo snapshot', () => {
    const component = renderer.create(<SpinnerComponent></SpinnerComponent>);
    const JSONComponent = component.toJSON();

    expect(JSONComponent).toMatchSnapshot();
  });
})
