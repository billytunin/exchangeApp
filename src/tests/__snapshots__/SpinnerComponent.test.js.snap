// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suite de tests del componente SpinnerComponent coincide con el Ãºltimo snapshot 1`] = `
<div
  className="spinner_container hide"
>
  <div
    className="spinner"
  />
</div>
`;
